app/javascript/components/Settings/PaymentsPage.tsx

```diff
@@ -279,6 +279,14 @@ const PaymentsPage = (props: Props) => {
     if (errorMessage) formRef.current?.scrollIntoView({ behavior: "smooth" });
   }, [errorMessage]);
 
+  React.useEffect(() => {
+    if (props.error_message) {
+      setErrorMessage({ message: props.error_message, code: props.error_code ?? null });
+    } else if (props.error_message === null) {
+      setErrorMessage(null);
+    }
+  }, [props.error_message, props.error_code]);
```

Comment:
We use useEffect to sync the error props (error_message and error_code) from the controller to local component state. When the controller re-renders with error props (via re_render_with_error), Inertia updates the page props, this effect runs, and it updates the local errorMessage state so the error banner displays. This keeps the component reactive to backend error state changes.